test-suite/testkit/tests/counter/main.rs

```diff
@@ -1018,3 +1019,35 @@ fn test_explorer_with_before_transactions_error() {
     assert_eq!(schema.counter.get(), Some(13));
     // ^-- The changes in `before_transactions` should be reverted.
 }
+
+/// Checks that `ExplorerApi` accepts valid transactions and discards transactions with incorrect instance ID.
+#[test]
+fn test_explorer_add_transaction_with_invalid_transaction() {
+    let (_testkit, api) = init_testkit();
+
+    // Send valid transaction.
+    let (pubkey, key) = crypto::gen_keypair();
+    let tx = Reset.sign(SERVICE_ID, pubkey, &key);
+    let data = hex::encode(tx.to_bytes());
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&json!({ "tx_body": data }))
+        .post::<TransactionResponse>("v1/transactions")
+        .unwrap();
+    assert_eq!(response.tx_hash, tx.object_hash());
+
+    // Send invalid transaction.
+    let (pubkey, key) = crypto::gen_keypair();
+    let tx = Reset.sign(SERVICE_ID + 1, pubkey, &key);
+    let data = hex::encode(tx.to_bytes());
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&json!({ "tx_body": data }))
+        .post::<TransactionResponse>("v1/transactions")
+        .unwrap_err();
```

Comment:
Nit: I'd suggest to use `expect_err` and `expect`, since the return value type is the topic of this test. With `expect_err`/`expect` we will receive a more meaningful messages if the test will fail.